doc/src/bond_cpp.md

```diff
@@ -185,6 +185,65 @@ particular protocol reader:
 
 See example: `examples/cpp/core/marshaling`.
 
+Schema evolution
+================
+
+Bond does not use explicit versioning to deal with changes to schemas (and the
+resulting types) over time. Instead, Bond supports certain schema evolution
+operations which allow the producer and consumer of Bond types to evolve
+without lockstep coordination.
+
+The following changes to a schema will never break compatibility across the wire:
+
+- Adding or removing an `optional` or `required_optional` field
+- Changing a field's type between `int32` and `enum`
+- Changing a field's type between `vector<T>` and `list<T>`
+- Changing a field's type between `blob` and `vector<uint8>` or `blob` and
+  `list<uint8>`
+- Changing a field's type between `T` and `bonded<T>`
+- Adding new enumeration constants that don't alter existing constants (beware
+  of implicit reordering)
+
+The following changes to a type are generally safe but require some
+consideration about how the change is rolled out:
+
+- Changing a field between `optional` and `required_optional` or between
+  `required_optional` and `required`. The `required_optional`. modifier
```

Comment:
Minor: it looks like there's an extraneous period after second `required_optional`.